pygmt/clib/session.py

```diff
@@ -1672,12 +1674,15 @@ def virtualfile_out(
             family, geometry = {
                 "dataset": ("GMT_IS_DATASET", "GMT_IS_PLP"),
                 "grid": ("GMT_IS_GRID", "GMT_IS_SURFACE"),
+                "image": ("GMT_IS_IMAGE", "GMT_IS_SURFACE"),
             }[kind]
-            with self.open_virtualfile(family, geometry, "GMT_OUT", None) as vfile:
+            with self.open_virtualfile(
+                family, geometry, "GMT_OUT|GMT_IS_REFERENCE", None
```

Comment:
For unknown reasons (likely upstream bug), using `GMT_OUT` crashes and `GMT_OUT|GMT_IS_REFERENCE` works for `image`. So better to use `GMT_OUT` for `dataset`/`grid` and `GMT_OUT|GMT_IS_REFERENCE` for `image`.

We also need to open an upstream issue and find out why.